### PR TITLE
[swift-3.0-preview-1] SimplifyCFG: fix crash in case an enum payload is an enum

### PR DESCRIPTION
**Pull request by eeckstein**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

fixes rdar://problem/26251856

### CODE REVIEW

**Comment by tkremenek** [comment]
@swift-ci test

File: test/SILOptimizer/simplify_cfg.sil [hunk 1]
```diff
@@ -2629,3 +2629,35 @@ bb3(%4 : $Optional<Int>):
 }
 
 
+// CHECK-LABEL: sil @dont_crash_on_enum_payload_is_enum
```
Comment (tkremenek) [inline test/SILOptimizer/simplify_cfg.sil]: Can we add a comment that indicates what is the purpose of this test?  The only indicator is this `dont_crash_on_enum_payload_is_enum` label, but that is a bit opaque.  Referencing the SR or Radar # would be good too.

Comment (eeckstein) [inline test/SILOptimizer/simplify_cfg.sil]: done. I updated the PR

**Comment by tkremenek** [comment]
@eeckstein looks like this won't merge cleanly.  GitHub says "this branch is out of sync with the base branch.

**Comment by eeckstein** [comment]
This is strange. I'll try to push again

**Comment by eeckstein** [comment]
@tkremenek It seems it can merge now

**Pull request closed without merging**